assets/src/models/createDetourMachine.ts

```diff
@@ -1,28 +1,134 @@
-import { setup } from "xstate"
+import { setup, assign } from "xstate"
+import { ShapePoint } from "../schedule"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+    context: {
```

Comment:
I think I mentioned in another PR, but I really like how clean the diff is when we augment the state machine. It makes this tool a lot easier to think about!